client/packages/system/src/Stock/ListView/ListView.tsx

```diff
@@ -179,6 +179,10 @@ const StockListComponent: FC = () => {
 
   const repackModalController = useToggle();
 
+  const stockLine = entity
+    ? data?.nodes.find(({ id }) => id === entity.id)
+    : undefined;
+
```

Comment:
This is the other one -- the `entity` is stored in a useState inside the Modal hook, and only gets updated on opening the modal. So if we derive `stockLine` from `data` rather than `entity` it will change when data changes :)

Reply:
Nice find... a bit of a shame though, that `entity` should make managing the modal data easier 🙈 

I reckon let's get this in, it works for now, though Andrei mentioned an idea last week that we could bring into the API hooks, that would help us do this more cleanly 🤓